topydo/commands/AddCommand.py

```diff
@@ -83,23 +98,47 @@ def add_dependencies(p_tag):
 
         self.todo.set_creation_date(date.today())
 
+    def get_todos_from_file(self):
+        if self.from_file == '-':
+            f = stdin
+        else:
+            f = codecs.open(self.from_file, 'r', encoding='utf-8')
+
+        todos = f.read().splitlines()
+
+        return todos
+
+    def _add_todo(self, p_todo_text):
+        todo_text = self._preprocess_input_todo(p_todo_text)
+        self.todo = self.todolist.add(todo_text)
```

Comment:
It looks like we can get rid of `self.todo` if you'd pass the `Todo` instance returned from `self.todolist.add()` to `self._postprocess_input_todo()` as a parameter.
Reason why I'd like to see it eliminated is that `AddCommand` is not necessarily dealing with a single todo item anymore, while the `todo` member attribute seems to imply otherwise. This makes reading the post-processing code a bit confusing.

Moreover, the preprocess and post-process methods do not need to be methods anymore. You can move them inside the `_add_todo` method, or outside the class.